site/src/routes/pins-replace.js

```diff
@@ -72,33 +73,19 @@ export async function pinsReplace(event, params) {
     )
   }
 
-  /** @type import('@nftstorage/ipfs-cluster').PinResponse | undefined */
-  let pin
-  /** @type import('../pinata-psa').Status */
-  let status = 'pinning'
-  let size = 0
-  try {
-    pin = await cluster.allocation(pinData.cid)
-  } catch (err) {
-    // if 404 we can continue, we have not pinned this CID
-    if (!err.response || err.response.status !== 404) {
-      throw err
-    }
-  }
-
+  const created = new Date().toISOString()
+  let pin = await pins.get(pinData.cid)
```

Comment:
This looks similar to what pins-add does and suggested refactor there would also enable code reuse.